platform/mock-radio.ts

```diff
@@ -0,0 +1,73 @@
+/*
+ * Mock radio for interfacing with mock devices
+ */
+
+export interface MockUDPListener {
+  onUDPMessage(msg: Buffer, rinfo: RemoteAddressInfo): void;
+}
+
+export class RemoteAddressInfo {
+  address: string;
+  family: string;
+  port: number;
+  size: number;
+  constructor(address: string, family: string, port: number, size: number) {
+    this.address = address;
+    this.family = family;
+    this.port = port;
+    this.size = size;
+  }
+}
+
+// Simulates a network with simple UDP messaging functionality
+export class MockNetwork {
+  udpListeners: Map<string, MockUDPListener[]>;
+
+  constructor() {
+    this.udpListeners = new Map<string, MockUDPListener[]>();
+  }
+
+  registerUDPListener(
+    listener: MockUDPListener,
+    port: number,
+    address: string
+  ) {
+    const key = address + ':' + port.toString();
+    if (this.udpListeners.has(key)) {
+      this.udpListeners[key].push(listener);
+      return;
+    }
+    this.udpListeners[key] = [listener];
+  }
+
+  public sendUDPMessage(
+    msg: Buffer,
+    port: number,
+    address: string,
+    fromPort: number,
+    fromAddress: string
+  ) {
+    const key = address + ':' + port.toString();
+    for (const listener of this.udpListeners[key]) {
+      const rinfo = {
+        port: fromPort,
+        address: fromAddress,
+      };
+      listener.onUDPMessage(msg, rinfo);
+    }
+  }
+}
+
+export class UDPDevice implements MockUDPListener {
+  private udpMessageAction: (msg: Buffer, rinfo: RemoteAddressInfo) => void;
+
+  onUDPMessage(msg: Buffer, rinfo: RemoteAddressInfo): void {
+    this.udpMessageAction(msg, rinfo);
+  }
+
+  setUDPMessageAction(
```

Comment:
Does this need to be dynamic? (i.e: would the given action associated to a listener change for a given instead?)

Reply:
I think this is also related to issue #27.